lib/rubygems/gemcutter_utilities.rb

```diff
@@ -267,6 +269,28 @@ def get_scope_params(scope)
     scope_params
   end
 
+  def get_mfa_params(email, password)
+    return {} unless self.host == Gem::DEFAULT_HOST
```

Comment:
Including this new code, this check is done already three times in this class. What about to refactor it into separated private method `default_host?`?